test/DurableTaskLifeCycleNotificationTest.cs

```diff
@@ -0,0 +1,361 @@
+            {
+                await host.StartAsync();
+                var extensionRegistry = (IExtensionRegistry)host.Services.GetService(typeof(IExtensionRegistry));
+                var extensionProviders = extensionRegistry.GetExtensions(typeof(IExtensionConfigProvider))
+                    .Where(x => x is DurableTaskExtension)
+                    .ToList();
+                if (extensionProviders.Any())
+                {
+                    var extension = (DurableTaskExtension)extensionProviders.First();
+                    var mock = new Mock<HttpMessageHandler>();
+                    mock.Protected()
+                        .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+                        .Returns((HttpRequestMessage request, CancellationToken cancellationToken) =>
+                            {
+                                Assert.True(request.Headers.Any(x => x.Key == "aeg-sas-key"));
+                                var values = request.Headers.GetValues("aeg-sas-key").ToList();
+                                Assert.Single(values);
+                                Assert.Equal(eventGridKey, values[0]);
+                                Assert.Equal(eventGridEndpoint, request.RequestUri.ToString());
+                                var json = request.Content.ReadAsStringAsync().GetAwaiter().GetResult();
+                                dynamic content = JsonConvert.DeserializeObject(json);
+                                foreach (dynamic o in content)
+                                {
+                                    Assert.Equal("1.0", o.dataVersion.ToString());
+                                    Assert.Equal(nameof(this.OrchestrationStartAndCompleted), o.data.HubName.ToString());
+                                    Assert.Equal(orchestratorFunctionNames[0], o.data.FunctionName.ToString());
+
+                                    if (callCount == 0)
+                                    {
+                                        Assert.Equal("durable/orchestrator/Running", o.subject.ToString());
+                                        Assert.Equal("orchestratorEvent", o.eventType.ToString());
+                                        Assert.Equal("0", o.data.EventType.ToString());
+                                    }
+                                    else if (callCount == 1)
+                                    {
+                                        Assert.Equal("durable/orchestrator/Completed", o.subject.ToString());
+                                        Assert.Equal("orchestratorEvent", o.eventType.ToString());
+                                        Assert.Equal("1", o.data.EventType.ToString());
+                                    }
+                                    else
+                                    {
+                                        throw new Exception("Call Count is Bad");
```

Comment:
How about `Assert.Fail`?

Reply:
Hello @cgillum ,

Since `Assert.Fail` is not available I used the recommended workaround - [xUnit.net alternative: Assert.True(false, "message")](http://xunit.github.io/docs/comparisons.html) and in we will have the following snippet:

```csharp
Assert.True(false, "Call Count is Bad");
```

Let me know if we need to modify it.

Thank you!